src/Symfony/Bundle/FrameworkBundle/Kernel/FrameworkKernel.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Kernel;
+
+use Symfony\Bundle\FrameworkBundle\Routing\RouteCollectionBuilder;
+use Symfony\Component\Config\Loader\LoaderInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\HttpKernel\MicroKernel;
+use Symfony\Component\Routing\Loader\RouteLoaderInterface;
+
+/**
+ * A kernel that adds some functionality that depends on FrameworkBundle.
+ *
+ * @author Ryan Weaver <ryan@knpuniversity.com>
+ */
+abstract class FrameworkKernel extends MicroKernel implements RouteLoaderInterface
```

Comment:
I would call this `MicroKernel` again. This is a common practice in Symfony code to add specific things to a class from another component (e.g. `Extension` in HttpKernel component).